src/frontend/src/helpers/Storage/Interface.ts

```diff
@@ -8,34 +8,42 @@ const defaultSettings: Settings = {
   tab_width: 2
 };
 
-export {AbstractStorage,
+export {AbstractStorage, AbstractWebStorage,
         File, FileID, FileBrief,
         Project, ProjectID, ProjectBrief,
         Settings, defaultSettings}
 
 
 abstract class AbstractStorage {
   // projects
-  public abstract async newProject(name: string): Promise<void>;
+  public abstract async newProject(name: string): Promise<ProjectID>;
   public abstract async getProject(proj: ProjectID): Promise<Project>;
   public abstract async getProjects(): Promise<ProjectBrief[]>;
   public abstract async deleteProject(proj: ProjectID): Promise<void>;
-  public abstract async getFiles(proj: ProjectID): Promise<FileBrief[]>;
+  public abstract async getProjectFiles(proj: ProjectID): Promise<FileBrief[]>;
   // files
   public abstract async newFile(proj: ProjectID, filename: string, contents?: string): Promise<FileID>;
   public abstract async readFile(file: FileID): Promise<File>;
   public abstract async writeFile(file: FileID, contents: string): Promise<void>;
-  public abstract async setFileToRun(proj: ProjectID, question: string, file: FileID): Promise<void>;
-  public abstract async getFileToRun(proj: ProjectID, question: string): Promise<FileID|undefined>;
   public abstract async renameFile(file: FileID, newName: string): Promise<void>;
   public abstract async deleteFile(file: FileID): Promise<void>;
+  // questions
+  public abstract async setFileToRun(proj: ProjectID, question: string, file: FileID): Promise<void>;
```

Comment:
setFileToRun/getFileToRun shouldn't take file IDs, they should take names as file IDs will refer to a particular revision of a file.